File: projects/tox4j/src/main/java/im/tox/tox4j/av/data/SampleCount.scala
package im.tox.tox4j.av.data

final case class SampleCount private (value: Int) extends AnyVal

case object SampleCount extends ((AudioLength, SamplingRate) => SampleCount) {

  override def apply(audioLength: AudioLength, samplingRate: SamplingRate): SampleCount = {
    new SampleCount(samplingRate.value / 1000 * audioLength.value.toMillis.toInt)
  }

}

File: lib/src/main/kotlin/im/tox/tox4j/av/data/SampleCount.kt
package im.tox.tox4j.av.data

import kotlin.jvm.JvmInline

@JvmInline
value class SampleCount(
    val value: Int,
) {
    constructor(
        audioLength: AudioLength,
        samplingRate: SamplingRate,
    ) : this((samplingRate.value / 1000 * audioLength.toMillis()).toInt())
}
